File: Practice/AOJ/ALDS1/ALDS1_03/ALDS1_03A/alds1_03a.py
arr = [int(n) if n.isnumeric() else n for n in list(map(str, input().split()))]
stack = []
for v in arr:
    if(type(v) is int):
        stack.append(v)
    else:
        # 取り出す順に注意
        b, a = stack.pop(), stack.pop()
        nx = a+b if v == "+" else a-b if v == "-" else a*b
        stack.append(nx)
print(stack.pop())

File: Practice/AOJ/ALDS1/ALDS1_03/ALDS1_03A/alds1_03a.rb
# 逆ポーランド記法ってスタックが話題になるときだけイキりだすよな...
arr = gets.chomp.split.map{|x| x.match(/\d/) ? x.to_i : x}
# evalは使わない

stack = []
arr.each do |item|
  if item.is_a?(Integer)
    stack << item
  else
    nx = case item
      when '+'
        stack.pop(2).inject(&:+)
      when '-'
        stack.pop(2).inject(&:-)
      when '*'
        stack.pop(2).inject(&:*)
      end
    stack << nx
  end
end
puts stack.pop